resources/assets/components/utilities/ScholarshipCard/GalleryBlockSignup.js

```diff
@@ -0,0 +1,52 @@
+import gql from 'graphql-tag';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
+import SecondaryButton from '../Button/SecondaryButton';
+import Spinner from '../../artifacts/Spinner/Spinner';
+
+export const CREATE_SIGNUP_MUTATION = gql`
+  mutation CampaignSignupMutation($campaignId: Int!) {
+    createSignup(campaignId: $campaignId) {
+      id
+    }
+  }
+`;
+const GalleryBlockSignup = ({ campaignId, path }) => {
+  const [handleSignup, { called, error }] = useMutation(
+    CREATE_SIGNUP_MUTATION,
+    {
+      variables: { campaignId },
+    },
+  );
+
+  if (called && !error) {
+    window.location = path;
+
+    return <Spinner className="flex justify-center p-4" />;
+  }
+
+  if (error) {
+    // TODO: show error state…
```

Comment:
I think adding in the error state here is all that's left! I tested the functionality (as a logged in user and not) on the review app and that all is working great!

```suggestion
     return <ErrorBlock error={error} />;
```

Reply:
@Vmack let me know if I can help with this!

Reply:
@katiecrane I just pushed the change, let me know if it looks good?